eva/binder/statement_binder.py

```diff
@@ -151,7 +152,36 @@ def _bind_delete_statement(self, node: DeleteTableStatement):
     @bind.register(CreateMaterializedViewStatement)
     def _bind_create_mat_statement(self, node: CreateMaterializedViewStatement):
         self.bind(node.query)
-        # Todo Verify if the number projected columns matches table
+        num_projected_columns = 0
+        # TODO we should fix the binder. All binded object should have the same interface.
+        for expr in node.query.target_list:
+            if expr.etype == ExpressionType.TUPLE_VALUE:
+                num_projected_columns += 1
+            elif expr.etype == ExpressionType.FUNCTION_EXPRESSION:
+                num_projected_columns += len(expr.output_objs)
+            else:
+                raise BinderError("Unsupported expression type {}.".format(expr.etype))
+
+        assert (
+            len(node.col_list) == 0 or len(node.col_list) == num_projected_columns
+        ), "Projected columns mismatch, expected {} found {}.".format(len(node.col_list), num_projected_columns)
+        binded_col_list = []
+        idx = 0
+        for expr in node.query.target_list:
```

Comment:
This logic is convoluted (too much nesting). It can be simplified. Can you point to the case where we cannot only rely on `objs`? They are required to be always correct.